src/ResourceManager/Insights/Commands.Insights/Metrics/GetAzureRmMetricCommand.cs

```diff
@@ -64,15 +65,47 @@ public class GetAzureRmMetricCommand : MonitorClientCmdletBase
         /// <summary>
         /// Gets or sets the starttime parameter of the cmdlet
         /// </summary>
+        [Parameter(ParameterSetName = GetAzureRmAMetricParamGroup, Mandatory = false, ValueFromPipelineByPropertyName = true, HelpMessage = "The start time of the query")]
         [Parameter(ParameterSetName = GetAzureRmAMetricFullParamGroup, Mandatory = false, ValueFromPipelineByPropertyName = true, HelpMessage = "The start time of the query")]
         public DateTime StartTime { get; set; }
 
         /// <summary>
         /// Gets or sets the endtime parameter of the cmdlet
         /// </summary>
+        [Parameter(ParameterSetName = GetAzureRmAMetricParamGroup, Mandatory = false, ValueFromPipelineByPropertyName = true, HelpMessage = "The end time of the query")]
         [Parameter(ParameterSetName = GetAzureRmAMetricFullParamGroup, Mandatory = false, ValueFromPipelineByPropertyName = true, HelpMessage = "The end time of the query")]
         public DateTime EndTime { get; set; }
 
+        /// <summary>
+        /// Gets or sets the top parameter of the cmdlet
+        /// </summary>
+        [Parameter(ParameterSetName = GetAzureRmAMetricFullParamGroup, Mandatory = false, ValueFromPipelineByPropertyName = true, HelpMessage = "The maximum number of records to retrieve (default:10), to be specified with $filter")]
+        public int? Top { get; set; }
```

Comment:
@pkamath13 @gucalder we should add the `ValidateRange` attribute to this parameter to restrict the values that a user could provide (such as negatives)